src/coreclr/utilcode/util.cpp

```diff
@@ -369,7 +369,13 @@ BYTE * ClrVirtualAllocWithinRange(const BYTE *pMinAddr,
     {
         NOTHROW;
         PRECONDITION(dwSize != 0);
-        PRECONDITION(flAllocationType == MEM_RESERVE);
+        PRECONDITION((flAllocationType & MEM_RESERVE) != 0);
+        PRECONDITION((flAllocationType & ~(
+                MEM_RESERVE
+#ifdef HOST_UNIX
```

Comment:
A nit -this doesn't do anything, since the precondition macros are empty for Unix.

Reply:
You also cannot have ifdef in there as it is inside of a macro expansion. I would just remove this and instead add a comment that on Unix, we can also have the `MEM_RESERVE_EXECUTABLE` set.

Reply:
Sure, does it look good now?